pkg/spi/resource/spi.go

```diff
@@ -0,0 +1,49 @@
+package resource
+
+import (
+	"encoding/json"
+
+	"github.com/docker/infrakit/pkg/spi"
+)
+
+// InterfaceSpec is the current name and version of the resource API.
+var InterfaceSpec = spi.InterfaceSpec{
+	Name:    "Resource",
+	Version: "0.1.0",
+}
+
+// Plugin is a vendor-agnostic API used to create and manage resources with an infrastructure provider.
+type Plugin interface {
```

Comment:
@nathanleclaire  that's a good suggestion.  Currently, the instance has the notion of an attachment, but that attachment / detachment is implicit inside the implementation of the instance plugin.  I wonder if it would better to make that more explicit and use the interface mechanism you described here.